### DISCUSSION TITLE
settings_parser.y error during setup, how to resolve

### DISCUSSION THREAD

**Discussion by Mathiaszero** [strongswan/strongswan#1800] [General]
I don't know if this is a bug or not, but I encountered an error regarding building my project.  During setup at the step after I ran "make," I get this:
`
a@xxxxxxxx:~/Downloads/strongswan$ make
make  all-recursive
make[1]: Entering directory '/home/a/Downloads/strongswan'
Making all in src
make[2]: Entering directory '/home/a/Downloads/strongswan/src'
Making all in .
make[3]: Entering directory '/home/a/Downloads/strongswan/src'
make[3]: Nothing to be done for 'all-am'.
make[3]: Leaving directory '/home/a/Downloads/strongswan/src'
Making all in include
make[3]: Entering directory '/home/a/Downloads/strongswan/src/include'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/home/a/Downloads/strongswan/src/include'
Making all in libstrongswan
make[3]: Entering directory '/home/a/Downloads/strongswan/src/libstrongswan'
/bin/bash ../../ylwrap settings/settings_parser.y y.tab.c settings/settings_parser.c y.tab.h `echo settings/settings_parser.c | sed -e s/cc$/hh/ -e s/cpp$/hpp/ -e s/cxx$/hxx/ -e s/c++$/h++/ -e s/c$/h/` y.output settings/settings_parser.output -- byacc -v -d 
byacc: e - line 68 of "/home/a/Downloads/strongswan/src/libstrongswan/settings/settings_parser.y", syntax error
%define parse.error verbose
^
make[3]: *** [Makefile:2195: settings/settings_parser.c] Error 1
make[3]: Leaving directory '/home/a/Downloads/strongswan/src/libstrongswan'
make[2]: *** [Makefile:539: all-recursive] Error 1
make[2]: Leaving directory '/home/a/Downloads/strongswan/src'
make[1]: *** [Makefile:603: all-recursive] Error 1
make[1]: Leaving directory '/home/a/Downloads/strongswan'
make: *** [Makefile:514: all] Error 2
`
Was trying to see if anyone else had this issue. I searched similar discussions on here, and did not find anything.  I was able to generate the Makefile after ./configure in the HACKING file, in Ubuntu 22.04. If no solution I will post on Issues then. Thanks in advance.

**Comment by tobiasbrunner** [comment]
Make sure you have the tools in HACKING installed (in this particular case `bison` and `flex`). Note that the same thing was basically reported in #1782 and #1276.

**Reply by Mathiaszero** [reply]
Thanks for the response. I will work on that in the meantime.
